src/lib.rs

```diff
@@ -800,11 +773,10 @@ impl Drop for Rav1dContext {
                     }
                 }
             }
-            let mut n_1: c_uint = 0 as c_int as c_uint;
-            while !(self.fc).is_null() && n_1 < self.n_fc {
-                let fc = &mut (*(self.fc).offset(n_1 as isize));
+            let fc_len = self.fc.len();
+            for fc in self.fc.iter_mut() {
                 let f = fc.data.get_mut().unwrap();
-                if self.n_fc > 1 as c_uint {
+                if fc_len > 1 {
                     let _ = mem::take(&mut f.lowest_pixel_mem); // TODO: remove when context is owned
                 }
```

Comment:
```suggestion
                let _ = mem::take(&mut f.lowest_pixel_mem); // TODO: remove when context is owned
```

We can avoid the separate `fc_len` variable by just unconditionally taking it.  When `fc.len() == 0`, `f.lowest_pixel_mem` is `Default::default()` already.

Reply:
Eh, we're going to rewrite all of this when the context is fully owned.